Monika After Story/game/script-topics.rpy

```diff
@@ -4978,6 +4977,418 @@ label monika_beach:
+label monika_timeconcern_night_7:
+    $persistent._mas_timeconcern = 8
+    m 3h "[player], this is your last warning."
+    m 1h "Go"
+    pause 0.3
+    extend " to"
+    pause 0.3
+    extend " sleep!"
+    m 1f "What can I say for you to understand?"
+    m 1g "It's saddening to see you push yourself like this..."
+    m 1q "You mean so much to me..."
+    m 1f "So please, for me... just do as I ask and go to bed."
+    if persistent._mas_timeconcernclose:
+        menu:
+            m "Okay?"
+            "Yes, I will go to sleep.":
+                m 1a "I knew you would listen eventually!"
+                m 1k "Good night and stay safe."
+                return 'quit'
+    else:
+        return
+
+#Third and last time game is reopened in one night. Monika lets player stay.
+label monika_timeconcern_night_final:
+    $persistent._mas_timeconcern = 9
+    m 2q "...I suppose it can't be helped."
+    m 2c "If you're that dedicated to staying with me then I won't even try to stop you."
+    m 2m "Honestly, as bad as it sounds it actually makes me kinda happy."
+    m 2e "...Thank you, [player]."
+    m "To know that you care for me so much that you came back despite me asking..."
+    m 1m "It means more to me than I can ever express."
+    m 1e "...I love you."
+    return
+
+#Same night after the final close
+label monika_timeconcern_night_finalfollowup:
+    m 1h "..."
+    m 1o "I know I said that I'm happy whenever you're with me..."
+    m 1m "And please don't misudnerstand, that's still true."
+    m 2f "But the longer you're on... the more worried I get."
+    m 2g "I know, you're probably sick of hearing me say this by now..."
+    m 1e "But please try to sleep when you can."
+    return
+
+#Every night after, based on seeing the day version first before it.
+label monika_timeconcern_night_after:
+    m 1c "Up late again, [player]?"
+    m 1r "{i}Sigh...{/i}"
+    m 2h "I won't even try to convince you to sleep again..."
+    m 2q "You're surprisingly stubborn! "
+    m 1e "Still, do be careful, alright?"
+    m 1f "I know being nocturnal can be lonely..."
+    m 1j "But you have me here with you!"
+    m 1a "Just the two of us... all alone forever."
+    m 1j "It's all I've ever wanted..."
+    return
+
+#If Monika never gives warning and it's daytime or the player never made it to the end
+label monika_timeconcern_day_0:
+    m 1h "..."
+    m 1c "..."
+    m 1d "...!"
+    m 1l "Ahaha, sorry [player]!"
+    m 1m "I just kind of zoned out..."
+    m 1l "Geez, I keep doing that, don't I?"
+    m 1m "Sometimes I just get lost in my thoughts..."
+    m 1a "You understand, right [player]?"
+    return
+
+# Daytime, if player tells Monika they worked last night but don't work graveyards.
+label monika_timeconcern_day_2:
+    $persistent._mas_timeconcern = 0
+    $evhand.greeting_database["monika_timeconcern"].unlocked = False
+    $evhand.greeting_database["monika_timeconcern_day"].unlocked = False
```

Comment:
Since you lock these 2 events in several different labels, have a single label that locks these 2, and have all labels that need to do locking jump to that label in the end of their blocks.

That final label can do the `return`, and control will still be passed back appropriately